### DISCUSSION TITLE
Issues with public sharing

### DISCUSSION THREAD

**Discussion by walterzilla** [devnoname120/epubviewer#22] [General] [answered]
NC AIO 29.0.4

Enabled apps:

```
  - activity: 2.21.1
  - cfg_share_links: 5.1.1
  - cloud_federation_api: 1.12.0
  - dav: 1.30.1
  - epubviewer: 1.6.3
  - federatedfilesharing: 1.19.0
  - files: 2.1.0
  - files_pdfviewer: 2.10.0
  - files_sharing: 1.21.0
  - files_trashbin: 1.19.0
  - logreader: 2.14.0
  - lookup_server_connector: 1.17.0
  - nextcloud-aio: 0.6.0
  - notifications: 2.17.0
  - oauth2: 1.17.0
  - provisioning_api: 1.19.0
  - serverinfo: 1.19.0
  - settings: 1.12.0
  - sharebymail: 1.19.0
  - text: 3.10.1
  - theming: 2.4.0
  - twofactor_backupcodes: 1.18.0
  - viewer: 2.3.0
  - workflowengine: 2.11.0
```

1) opening a public share link (read only or whatever options, doesn't matter) to an epub file as unauthenticated user results in error:

![image](https://github.com/user-attachments/assets/7a6ff6c3-a0f8-400e-873f-a246770c1613)

NC log: `Cannot assign null to property OCA\Epubviewer\Controller\PageController::$userId of type string` 

Beside the error, maybe it could be helpful knowing that opening the link generates a file download notification anyway.

2) selecting an epub file from a public shared folder (as authenticated or unauthenticated user) always triggers the download instead of opening/preview as normally occurs with other supported file types, in fact nothing happens at all if download is disabled:

![test](https://github.com/user-attachments/assets/f1250349-0476-46c1-8c50-a4cb98a2f718)

**Comment by devnoname120** [comment]
With which Nextcloud version did the internal server error happen? I don't mean Nextcloud 29, I mean something like Nextcloud 29.1.2. Be precise please or I won't be able to fix the bug.

**Comment by walterzilla** [comment]
Guess you missed it in my opening post:

> NC AIO 29.0.4

BTW bug was appearing in previous releases too.
Thanks

**Reply by devnoname120** [reply]
> BTW bug was appearing in previous releases too.

> With which Nextcloud version did the internal server error happen?

**Reply by walterzilla** [reply]
> With which Nextcloud version did the internal server error happen?

Don't remember anymore...sure isn't working as expected with 29.0.4 (and 29.0.5, tested few days ago),

**Reply by devnoname120** [reply]
You told me that the “bug was appearing in previous releases too” so it didn't start happening with Nextcloud AIO 29.0.4?

**Reply by walterzilla** [reply]
> You told me that the “bug was appearing in previous releases too” so it didn't start happening with Nextcloud AIO 29.0.4?

Sorry I misunderstood your previous reply, now post edited...anyway I confirm it was already there (docker or not) but I just hadn't thought to report it.

**Comment by Atrium0804** [comment]
I get the same error too when I share a epub and the recipient opens the public link
NC 30.0.1
Epub viewer 1.6,.3

[index] Fout: Cannot assign null to property OCA\Epubviewer\Controller\PageController::$userId of type string
	GET /apps/epubviewer/?file=%2Fs%2FZzcK43bQ9NksYwW%2Fdownload&type=application%2Fepub%2Bzip
	from <ip address> by -- at 23 okt 2024 11:47:53

**Comment by rportnoy2004** [comment]
same problem for me here on NC 27.1.11

**Comment by devnoname120** [comment, marked as answer]
Fixed in https://github.com/devnoname120/epubviewer/releases/tag/1.6.4
